### DISCUSSION TITLE
Gray background for generic code cells

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#2497] [Q&A] [answered]
Code cells that can be executed (e.g. Python or R) are rendered with a gray background. Other code cells have a white background. Is it possible to change this to gray so that all code cells look the same (except of course for syntax highlighting)?

Example:
````
---
title: "Untitled"
format: html
---

This:

```
pip list
```

looks different from this:

```{python}
print("Hello")
```
````

Which renders as:
<img width="849" alt="Screen Shot 2022-09-19 at 15 24 25" src="https://user-images.githubusercontent.com/4377312/191027621-c8737999-2659-4a9e-9df6-9127e129cc2c.png">

I would like the first code cell at the top ("pip list") to have a gray background (or rather, the same style as the Python code cell). Otherwise, it's really hard to distinguish between inputs and outputs. Is this possible?

**Comment by rgaiacs** [comment]
@cbrnr If you are using RMarkdown, you can replace

``````
```
pip list
```
``````

with

``````
```{bash}
#| eval: false

pip list
```
``````

With Jupyter, it's not possible because Jupyter only allows one kernel per Notebook.

**Reply by cbrnr** [reply]
Wow, I didn't know that! This works great! Still, do you (or anyone else) know if this is also possible with Jupyter? Some custom CSS maybe?

**Reply by cbrnr** [reply]
In addition, using `knitr` requires the `reticulate` package. If I don't want to use it, I can use the

```
#| python.reticulate: false
```

chunk option. Do you know how I can set this as a global option?

**Reply by cbrnr** [reply]
OK, to answer my last question, this is what needs to go into the header:

```
knitr:
  opts_chunk:
    python.reticulate: false
```

**Comment by cscheid** [comment, marked as answer]
Also,

````
```bash
pip list
```
````

will set the formatting to match that of

````
```{bash}
pip list
```
````

but without actually executing it

**Reply by cbrnr** [reply]
That's great, and it even works with the Jupyter engine. I didn't know that this is basically a shortcut for `#| eval: false`, good to know! Is this documented?

**Reply by cderv** [reply]
It is not a shortcut to `#| eval : false`, it is just basic Markdown syntax for code block:
* Doc in quarto: https://quarto.org/docs/authoring/markdown-basics.html#source-code
* Doc in Pandoc's Markdown: https://pandoc.org/MANUAL.html#extension-fenced_code_attributes

Setting a language adds the information for pandoc highlighting mechanism to work. It could be written this way to 
````
```{.bash}
pip list
```
````

Difference is that this is not a computational chunk, and no engine is required to render a document with only verbatim code chunk. Setting chunks will `#| eval: false` will still require an engine. Use the former if you don't need execution.

**Reply by cbrnr** [reply]
Thank you so much for the explanation! This is extremely useful to know!
